blockchain/reactor_test.go

```diff
@@ -0,0 +1,204 @@
+package blockchain_test
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"testing"
+
+	"github.com/tendermint/ed25519"
+	"github.com/tendermint/go-crypto"
+	wire "github.com/tendermint/go-wire"
+	"github.com/tendermint/tendermint/blockchain"
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/consensus"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/proxy"
+	sm "github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/state/txindex"
+	"github.com/tendermint/tendermint/state/txindex/kv"
+	"github.com/tendermint/tendermint/state/txindex/null"
+	"github.com/tendermint/tendermint/types"
+	cmn "github.com/tendermint/tmlibs/common"
+	"github.com/tendermint/tmlibs/db"
+	"github.com/tendermint/tmlibs/log"
+)
+
+type bcBlockRequestMessage struct {
+	Height int
+}
+
+func init() {
+	brm := &bcBlockRequestMessage{40}
+	buf := new(bytes.Buffer)
+	var n int
+	var err error
+	wire.WriteBinary(brm, buf, &n, &err)
+	fmt.Printf("% x\n", buf.Bytes())
+}
+
+func newBlockchainReactor(logger log.Logger, blockNumbers ...int) (*blockchain.BlockchainReactor, *sm.State) {
```

Comment:
We shouldn't need such complex setup. See how we setup consensus tests for reference.

We make whole networks in process here: https://github.com/tendermint/tendermint/blob/master/consensus/common_test.go#L357

And just the state machine here: https://github.com/tendermint/tendermint/blob/master/consensus/common_test.go#L232

We can hard code an inproc dummy app. No need for handshaker. No need for tx indexing. No need to log if we're a validator. Fast sync should be on. etc

Reply:
Awesome, thank you @ebuchman for the reference. Yeah, it's mostly out of lack of familiarity, and it yeah I was also telling myself that y'all def wrote simpler test setups, but TIL!